src/js/component.js

```diff
@@ -630,11 +708,17 @@ class Component {
    * you need to call `myComponent.trigger(el, 'dispose')` on it to clean up
    * references to it and allow the browser to garbage collect it.
    *
-   * @param  {String|Component} first   The event type or other component
-   * @param  {Function|String}      second  The event handler or event type
-   * @param  {Function}             third   The event handler
+   * @param {string|Component} first
+   *        The event type or other component
+   *
+   * @param {Function|string} second
+   *        The event handler or event type
+   *
+   * @param {Function} third
+   *        The event handler
+   *
    * @return {Component}
-   * @method on
+   *         a reference to the component this function was called with
```

Comment:
well sometime `@return {Component}` will return a component that does not refer to the one that the function was called on. Although I do agree we need some sort of convention for it.